pkg/config/url.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/openshift/odo/pkg/localConfigProvider"
+	"github.com/openshift/odo/pkg/util"
+)
+
+// GetPorts returns the ports, returns default if nil
+func (lc *LocalConfig) GetPorts() []string {
+	if lc.componentSettings.Ports == nil {
+		return nil
```

Comment:
>  Also, why not just return lc.componentSettings.Ports and do the check in the calling function?

This will result in duplicate checks in the calling functions.

> This if condition seems unnecessary to me because in both cases, we're just returning the value in lc.componentSettings.Ports

Without the if statement, the function can result in
```
panic: runtime error: invalid memory address or nil pointer dereference
```
as the value of Ports maybe nil.

Reply:
Thanks for that explanation. Makes sense. I stand corrected. :bow: